src/test/kotlin/game/racing/domain/RacingManagerTest.kt

```diff
@@ -1,21 +1,39 @@
 package game.racing.domain
 
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
 import org.junit.jupiter.params.ParameterizedTest
-import org.junit.jupiter.params.provider.ValueSource
+import org.junit.jupiter.params.provider.MethodSource
 
 internal class RacingManagerTest {
     @ParameterizedTest
-    @ValueSource(strings = ["WIN,TEST2,TEST3", "TEST1,TEST2,WIN,TEST4", "TEST1,WIN,TEST3,TEST4,TEST5"])
-    fun `우승자를 반환한다`(carNames: String) {
-        val cars = carNames.split(",").toTypedArray().map { Car(it) }
+    @MethodSource("provideCarList")
+    fun `우승자를 반환한다`(cars: List<Car>) {
+        // when
         val racingManager = RacingManager(cars)
-        cars.forEach {
-            if (it.name == "WIN") {
-                it.moveOrStayBySpeed(Car.MIN_MOVE_THRESHOLD)
-            }
+        cars.filter { it.name == "WIN" }.forEach {
+            it.moveOrStayBySpeed(Car.MIN_MOVE_THRESHOLD)
         }
 
+        // then
         assertThat(racingManager.determineWinner()).allMatch { it.position == Car.MIN_POSITION + 1 }.allMatch { it.name == "WIN" }
     }
+
+    @Test
+    fun `자동차가 움직인다`() {
+        val cars = arrayOf(Car("TEST1"), Car("TEST2"), Car("TEST3"))
+        val racingManager = RacingManager(cars.toList())
+        racingManager.moveCars { Car.MIN_MOVE_THRESHOLD }
+
+        assertThat(cars).allMatch { it.position == Car.MIN_POSITION + 1 }
```

Comment:
테스트는 의도한대로 나오는지를 검증하는 것이 핵심이라고 생각해요. 😃
다소 검증단이 투박(?)해 보일 수 있지만 의도한 것은 자동차 3대가 모두 포지션이 1이 되는 것이니 아래와 같이 작성하는 것이 더 좋다고 생각합니다. 🤗 

```suggestion
        assertThat(cars[0].position).isEqualTo(1)
        assertThat(cars[1].position).isEqualTo(1)
        assertThat(cars[2].position).isEqualTo(1)
```

만약 지금처럼 테스트 코드를 작성하게되면 추후 기능이 변경됐는데도 모르고 지나칠 수 있어요. 😅
그래서 비즈니스 로직에서는 하드코딩을 지양해야하는 반면에 테스트 코드에서는 오히려 지향하는 것이 좋은 것 같습니다. 😉